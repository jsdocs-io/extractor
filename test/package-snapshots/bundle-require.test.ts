import { describe, it } from "vitest";
import { expectPackageAPIToMatchSnapshot } from "../helpers/expect-package-api-to-match-snapshot";

describe("bundle-require", () => {
  const name = "bundle-require";

  it("2.1.2", async () => {
    await expectPackageAPIToMatchSnapshot({ name, version: "2.1.2" });
  });
});
