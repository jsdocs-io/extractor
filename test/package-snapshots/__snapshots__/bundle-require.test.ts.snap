// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`bundle-require > 2.1.2 1`] = `
{
  "declarations": {
    "classes": [],
    "enums": [],
    "functions": [
      {
        "docs": [],
        "id": "bundleRequire",
        "kind": "FunctionDeclaration",
        "name": "bundleRequire",
        "signature": "bundleRequire: (
    options: Options
) => Promise<{ mod: any; dependencies: string[] }>;",
        "source": {
          "filename": "dist/index.d.ts",
          "line": 46,
          "unpkgURL": "https://unpkg.com/browse/bundle-require@2.1.2/dist/index.d.ts#L46",
          "url": undefined,
        },
        "type": "(options: Options) => Promise<{    mod: any;    dependencies: string[];}>",
      },
    ],
    "interfaces": [
      {
        "docs": [],
        "id": "Options",
        "kind": "InterfaceDeclaration",
        "members": {
          "callSignatures": [],
          "constructSignatures": [],
          "indexSignatures": [],
          "methods": [],
          "properties": [
            {
              "docs": [
                "/**
     * esbuild options
     */",
              ],
              "id": "Options.esbuildOptions",
              "isOptional": true,
              "isReadonly": false,
              "kind": "InterfacePropertyDeclaration",
              "name": "esbuildOptions",
              "signature": "esbuildOptions?: BuildOptions;",
              "source": {
                "filename": "dist/index.d.ts",
                "line": 31,
                "unpkgURL": "https://unpkg.com/browse/bundle-require@2.1.2/dist/index.d.ts#L31",
                "url": undefined,
              },
              "type": "BuildOptions",
            },
            {
              "docs": [
                "/**
     * The filepath to bundle and require
     */",
              ],
              "id": "Options.filepath",
              "isOptional": false,
              "isReadonly": false,
              "kind": "InterfacePropertyDeclaration",
              "name": "filepath",
              "signature": "filepath: string;",
              "source": {
                "filename": "dist/index.d.ts",
                "line": 21,
                "unpkgURL": "https://unpkg.com/browse/bundle-require@2.1.2/dist/index.d.ts#L21",
                "url": undefined,
              },
              "type": "string",
            },
            {
              "docs": [
                "/**
     * Get the path to the output file
     * By default we simply replace the extension with \`.bundled.js\`
     */",
              ],
              "id": "Options.getOutputFile",
              "isOptional": true,
              "isReadonly": false,
              "kind": "InterfacePropertyDeclaration",
              "name": "getOutputFile",
              "signature": "getOutputFile?: (filepath: string) => string;",
              "source": {
                "filename": "dist/index.d.ts",
                "line": 36,
                "unpkgURL": "https://unpkg.com/browse/bundle-require@2.1.2/dist/index.d.ts#L36",
                "url": undefined,
              },
              "type": "(filepath: string) => string",
            },
            {
              "docs": [
                "/**
     * Enable watching and call the callback after each rebuild
     */",
              ],
              "id": "Options.onRebuild",
              "isOptional": true,
              "isReadonly": false,
              "kind": "InterfacePropertyDeclaration",
              "name": "onRebuild",
              "signature": "onRebuild?: (ctx: {
    err?: BuildFailure;
    mod?: any;
    dependencies?: string[];
}) => void;",
              "source": {
                "filename": "dist/index.d.ts",
                "line": 40,
                "unpkgURL": "https://unpkg.com/browse/bundle-require@2.1.2/dist/index.d.ts#L40",
                "url": undefined,
              },
              "type": "(ctx: {    err?: BuildFailure;    mod?: any;    dependencies?: string[];}) => void",
            },
            {
              "docs": [
                "/**
     * The \`require\` function that is used to load the output file
     * Default to the global \`require\` function
     * This function can be asynchronous, i.e. returns a Promise
     */",
              ],
              "id": "Options.require",
              "isOptional": true,
              "isReadonly": false,
              "kind": "InterfacePropertyDeclaration",
              "name": "require",
              "signature": "require?: RequireFunction;",
              "source": {
                "filename": "dist/index.d.ts",
                "line": 27,
                "unpkgURL": "https://unpkg.com/browse/bundle-require@2.1.2/dist/index.d.ts#L27",
                "url": undefined,
              },
              "type": "RequireFunction",
            },
          ],
        },
        "name": "Options",
        "signature": "interface Options {}",
        "source": {
          "filename": "dist/index.d.ts",
          "line": 17,
          "unpkgURL": "https://unpkg.com/browse/bundle-require@2.1.2/dist/index.d.ts#L17",
          "url": undefined,
        },
      },
    ],
    "namespaces": [],
    "typeAliases": [
      {
        "docs": [],
        "id": "RequireFunction",
        "kind": "TypeAliasDeclaration",
        "name": "RequireFunction",
        "signature": "type RequireFunction = (
    outfile: string,
    ctx: {
        format: 'cjs' | 'esm';
    }
) => any;",
        "source": {
          "filename": "dist/index.d.ts",
          "line": 14,
          "unpkgURL": "https://unpkg.com/browse/bundle-require@2.1.2/dist/index.d.ts#L14",
          "url": undefined,
        },
      },
    ],
    "variables": [
      {
        "docs": [
          "/**
 * Dynamically import files. It will make sure it's not being compiled away by TS/Rollup.
 *
 * As a temporary workaround for Jest's lack of stable ESM support, we fallback to require
 * if we're in a Jest environment.
 * See https://github.com/vitejs/vite/pull/5197#issuecomment-938054077
 *
 * @param file File path to import.
 */",
        ],
        "id": "dynamicImport",
        "kind": "VariableDeclaration",
        "name": "dynamicImport",
        "signature": "const dynamicImport: RequireFunction;",
        "source": {
          "filename": "dist/index.d.ts",
          "line": 12,
          "unpkgURL": "https://unpkg.com/browse/bundle-require@2.1.2/dist/index.d.ts#L12",
          "url": undefined,
        },
        "type": "RequireFunction",
        "variableKind": "const",
      },
    ],
  },
  "files": [
    {
      "filename": "dist/index.d.ts",
      "isIndexFile": true,
      "unpkgURL": "https://unpkg.com/browse/bundle-require@2.1.2/dist/index.d.ts",
      "url": undefined,
    },
  ],
  "overview": undefined,
}
`;
